### PR TITLE
[ONEM-20245] LISA getProgress impl; debugs reworked

### PR DESCRIPTION
**Pull request by szymon-polke-red**

### CODE REVIEW

File: LISA/Downloader.cpp [hunk 1]
```diff
@@ -181,11 +186,23 @@ int Downloader::curlProgressCb(void* userData,
 
 bool Downloader::onProgress(long dlTotal, long dlNow)
 {
-    TRACE_L1("downloaded: %ld/%ld", dlNow, dlTotal);
-    // TODO progress notification
+    Progress newProgres = {dlTotal, dlNow};
+    if (newProgres != progress) {
+        progress = newProgres;
```
Comment (piotr-marcinkowski-red) [inline LISA/Downloader.cpp:191]: This assignment should probably by synchronized - LockGuard lock(taskMutex);

**Review by piotr-marcinkowski-red — APPROVED** [review]

File: LISA/Downloader.cpp (hunk 1 above)
Comment (szymon-polke-red) [reply LISA/Downloader.cpp:191]: synchronization is done in Executor.cpp in setProgress

File: LISA/Debug.h [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+#define LOG_INTERNAL(...)  \
+    std::ostringstream os; \
+    lisaInternalMakeLogMessage(os, __VA_ARGS__); \
+    std::string str = os.str();
+
+/** INTERNAL HELPERS END ***************************************************************/
+
+
+#define INFO(...) { \
```
Comment (mariusz-sikora-red) [inline LISA/Debug.h:63]: Shouldn't be 'do while(0)'?

File: LISA/Executor.cpp [hunk 3]
```diff
@@ -233,52 +245,87 @@ void Executor::doInstall(std::string type,
...
+std::ostream& operator<<(std::ostream& out, const Executor::Task& task)
+{
+    return out << "task[" << task.handle << "]";
+}
+
+std::ostream& operator<<(std::ostream& out, Executor::OperationStage stage)
+{
+    const std::array<const char*, enumToInt(Executor::OperationStage::COUNT)> stages = {{
+            "DOWNLOADING",
+            "UNTARING",
```
Comment (mariusz-sikora-red) [inline LISA/Executor.cpp:324]: Maybe EXTRACT? Or it will be always tar archive?

Comment (szymon-polke-red) [reply LISA/Executor.cpp:324]: Hard to say how it will change in the future ;) for now docs explicitly mention untar, so let's leave it as it is.

**Review by mariusz-sikora-red — APPROVED** [review]

**Pull request merged**